tikv-client/src/main/java/com/pingcap/tikv/key/CommonHandle.java

```diff
@@ -53,11 +53,20 @@ public static CommonHandle newCommonHandle(
     CodecDataOutput cdo = new CodecDataOutput();
     for (int i = 0; i < data.length; i++) {
       if (dataTypes[i].getType().equals(MySQLType.TypeTimestamp)) {
-        long milliseconds = ((Timestamp) data[i]).getTime();
+        // When writing `Timestamp`, it will pass `Timestamp` object.
+        // When indexScan or tableScan, it will pass `Long` object.
+        // It's a compromise here since we don't have a good way to make them consistent.
+        long milliseconds;
```

Comment:
use pattern match may be better
- for Timestamp use getTime() when
- for Long /1000
- for other type throw exception